File: Scheduler/SchedulerMultipleResourcesEditDialog/SchedulerMultipleResourcesEditDialogVB/MultipleResourcesDialog.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Telerik.WinControls.UI.Scheduler.Dialogs
Imports Telerik.WinControls.UI

Public Class MultipleResourcesDialog
    Inherits EditAppointmentDialog
    Private listResources As Telerik.WinControls.UI.RadListControl

    Public Sub New()
        InitializeComponent()
    End Sub

    Protected Overrides Sub LoadSettingsFromEvent(ByVal sourceEvent As IEvent)
        MyBase.LoadSettingsFromEvent(sourceEvent)

        If sourceEvent.ResourceId IsNot Nothing Then
            For Each item As RadListDataItem In Me.listResources.Items
                If sourceEvent.ResourceIds.Contains(TryCast(item.Value, EventId)) Then
                    item.Selected = True
                Else
                    item.Selected = False
                End If
            Next

            If Me.SchedulerData.GroupType = GroupType.None AndAlso sourceEvent.ResourceIds.Count > 1 Then
                Me.listResources.SelectedIndex = 1

            End If
        End If
    End Sub

    Protected Overrides Sub LoadResources()
        MyBase.LoadResources()

        If Me.SchedulerData Is Nothing Then
            Return
        End If

        Dim resourceStorage As ISchedulerStorage(Of IResource) = Me.SchedulerData.GetResourceStorage()
        Me.listResources.Items.Clear()

        If Me.SchedulerData.GroupType = GroupType.None Then
            Dim item As New RadListDataItem("None")
            item.Value = -1
            Me.listResources.Items.Add(item)
        End If

        For Each resource As IResource In resourceStorage
            If resource.Visible Then
                Dim item As New RadListDataItem(resource.Name)
                item.Value = resource.Id
                Me.listResources.Items.Add(item)
            End If
        Next

        If Me.listResources.Items.Count > 0 Then
            Me.listResources.SelectedIndex = 0
        End If
    End Sub

    Protected Overrides Sub ApplySettingsToEvent(ByVal targetEvent As IEvent)
        MyBase.ApplySettingsToEvent(targetEvent)

        If Me.listResources.SelectedIndex >= 0 Then
            Dim selectedItem As RadListDataItem = listResources.Items(Me.listResources.SelectedIndex)

            If selectedItem.Text.Equals("None") Then
                If targetEvent.ResourceIds.Count > 0 Then
                    targetEvent.ResourceIds.Clear()
                End If
            Else
                targetEvent.ResourceIds.Clear()

                For Each item As RadListDataItem In listResources.SelectedItems
                    Dim resourceId As EventId = TryCast(item.Value, EventId)
                    If Not targetEvent.ResourceIds.Contains(resourceId) Then
                        targetEvent.ResourceIds.Add(resourceId)
                    End If
                Next
            End If
        End If
    End Sub

    Private Sub InitializeComponent()
        Me.listResources = New Telerik.WinControls.UI.RadListControl()
        DirectCast(Me.labelSubject, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.labelLocation, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.labelBackground, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.txtSubject, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.txtLocation, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.cmbBackground, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.labelStartTime, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.labelEndTime, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.dateStart, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.timeStart, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.dateEnd, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.timeEnd, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.chkAllDay, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.labelResource, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.cmbResource, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.labelStatus, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.cmbShowTimeAs, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.textBoxDescription, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.buttonOK, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.buttonCancel, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.buttonDelete, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.buttonRecurrence, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.errorProvider, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.radSeparator1, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.radSeparator2, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.radSeparator3, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me.listResources, System.ComponentModel.ISupportInitialize).BeginInit()
        DirectCast(Me, System.ComponentModel.ISupportInitialize).BeginInit()
        Me.SuspendLayout()
        ' 
        ' cmbBackground
        ' 
        ' 
        ' 
        ' 
        Me.cmbBackground.RootElement.AutoSizeMode = Telerik.WinControls.RadAutoSizeMode.WrapAroundChildren
        Me.cmbBackground.Size = New System.Drawing.Size(180, 20)
        ' 
        ' dateStart
        ' 
        Me.dateStart.Value = New System.DateTime(2011, 5, 17, 11, 22, 29, _
         614)
        ' 
        ' timeStart
        ' 
        Me.timeStart.Value = New System.DateTime(2011, 5, 17, 11, 22, 29, _
         614)
        ' 
        ' dateEnd
        ' 
        Me.dateEnd.Value = New System.DateTime(2011, 5, 17, 11, 22, 29, _
         614)
        ' 
        ' timeEnd
        ' 
        Me.timeEnd.Value = New System.DateTime(2011, 5, 17, 11, 22, 29, _
         614)
        ' 
        ' chkAllDay
        ' 
        ' 
        ' 
        ' 
        Me.chkAllDay.RootElement.ForeColor = System.Drawing.Color.FromArgb(CInt(CByte(5)), CInt(CByte(5)), CInt(CByte(5)))
        ' 
        ' cmbResource
        ' 
        Me.cmbResource.Location = New System.Drawing.Point(345, 169)
        Me.cmbResource.Size = New System.Drawing.Size(180, 20)
        Me.cmbResource.Visible = False
        ' 
        ' cmbShowTimeAs
        ' 
        ' 
        ' 
        ' 
        Me.cmbShowTimeAs.RootElement.AutoSizeMode = Telerik.WinControls.RadAutoSizeMode.WrapAroundChildren
        Me.cmbShowTimeAs.Size = New System.Drawing.Size(178, 20)
        ' 
        ' textBoxDescription
        ' 
        Me.textBoxDescription.Location = New System.Drawing.Point(7, 322)
        ' 
        ' 
        ' 
        Me.textBoxDescription.RootElement.StretchVertically = True
        Me.textBoxDescription.Size = New System.Drawing.Size(519, 116)
        ' 
        ' buttonOK
        ' 
        Me.buttonOK.Location = New System.Drawing.Point(6, 444)
        ' 
        ' buttonCancel
        ' 
        Me.buttonCancel.Location = New System.Drawing.Point(87, 444)
        ' 
        ' buttonDelete
        ' 
        Me.buttonDelete.Location = New System.Drawing.Point(168, 444)
        ' 
        ' buttonRecurrence
        ' 
        Me.buttonRecurrence.Location = New System.Drawing.Point(249, 444)
        ' 
        ' radSeparator3
        ' 
        Me.radSeparator3.Location = New System.Drawing.Point(6, 306)
        ' 
        ' listResources
        ' 
        Me.listResources.CaseSensitiveSort = True
        Me.listResources.ItemHeight = 18
        Me.listResources.Location = New System.Drawing.Point(79, 169)
        Me.listResources.Name = "listResources"
        Me.listResources.SelectionMode = System.Windows.Forms.SelectionMode.MultiSimple
        Me.listResources.Size = New System.Drawing.Size(180, 131)
        Me.listResources.TabIndex = 25
        Me.listResources.Text = "radListControl1"
        ' 
        ' MultipleResourcesDialog
        ' 
        Me.ClientSize = New System.Drawing.Size(537, 472)
        Me.Controls.Add(Me.listResources)
        Me.Name = "MultipleResourcesDialog"
        ' 
        ' 
        ' 
        Me.RootElement.ApplyShapeToControl = True
        Me.RootElement.MinSize = New System.Drawing.Size(539, 365)
        Me.Text = "Edit Appointment"
        Me.Controls.SetChildIndex(Me.listResources, 0)
        Me.Controls.SetChildIndex(Me.radSeparator3, 0)
        Me.Controls.SetChildIndex(Me.cmbResource, 0)
        Me.Controls.SetChildIndex(Me.labelSubject, 0)
        Me.Controls.SetChildIndex(Me.labelLocation, 0)
        Me.Controls.SetChildIndex(Me.labelBackground, 0)
        Me.Controls.SetChildIndex(Me.txtSubject, 0)
        Me.Controls.SetChildIndex(Me.txtLocation, 0)
        Me.Controls.SetChildIndex(Me.cmbBackground, 0)
        Me.Controls.SetChildIndex(Me.labelStartTime, 0)
        Me.Controls.SetChildIndex(Me.labelEndTime, 0)
        Me.Controls.SetChildIndex(Me.dateStart, 0)
        Me.Controls.SetChildIndex(Me.timeStart, 0)
        Me.Controls.SetChildIndex(Me.dateEnd, 0)
        Me.Controls.SetChildIndex(Me.timeEnd, 0)
        Me.Controls.SetChildIndex(Me.chkAllDay, 0)
        Me.Controls.SetChildIndex(Me.labelResource, 0)
        Me.Controls.SetChildIndex(Me.labelStatus, 0)
        Me.Controls.SetChildIndex(Me.cmbShowTimeAs, 0)
        Me.Controls.SetChildIndex(Me.textBoxDescription, 0)
        Me.Controls.SetChildIndex(Me.buttonOK, 0)
        Me.Controls.SetChildIndex(Me.buttonCancel, 0)
        Me.Controls.SetChildIndex(Me.buttonDelete, 0)
        Me.Controls.SetChildIndex(Me.buttonRecurrence, 0)
        Me.Controls.SetChildIndex(Me.radSeparator1, 0)
        Me.Controls.SetChildIndex(Me.radSeparator2, 0)
        DirectCast(Me.labelSubject, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.labelLocation, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.labelBackground, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.txtSubject, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.txtLocation, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.cmbBackground, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.labelStartTime, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.labelEndTime, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.dateStart, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.timeStart, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.dateEnd, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.timeEnd, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.chkAllDay, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.labelResource, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.cmbResource, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.labelStatus, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.cmbShowTimeAs, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.textBoxDescription, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.buttonOK, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.buttonCancel, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.buttonDelete, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.buttonRecurrence, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.errorProvider, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.radSeparator1, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.radSeparator2, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.radSeparator3, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me.listResources, System.ComponentModel.ISupportInitialize).EndInit()
        DirectCast(Me, System.ComponentModel.ISupportInitialize).EndInit()
        Me.ResumeLayout(False)
        Me.PerformLayout()

    End Sub
End Class

File: Scheduler/SchedulerMultipleResourcesEditDialog/SchedulerMultipleResourcesEditDialog/MultipleResourcesDialog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI.Scheduler.Dialogs;
using Telerik.WinControls.UI;

namespace SchedulerMultipleResourcesEditDialog
{
    public class MultipleResourcesDialog : EditAppointmentDialog
    {
        private Telerik.WinControls.UI.RadListControl listResources;

        public MultipleResourcesDialog()
        {
            InitializeComponent();
        }

        protected override void LoadSettingsFromEvent(IEvent sourceEvent)
        {
            base.LoadSettingsFromEvent(sourceEvent);

            if (sourceEvent.ResourceId != null)
            { 
                foreach (RadListDataItem item in this.listResources.Items)
                {
                    if (sourceEvent.ResourceIds.Contains(item.Value as EventId))
                    {
                        item.Selected = true;
                    }
                    else
                    {
                        item.Selected = false;
                    }
                }

                if (this.schedulerData.GroupType == GroupType.None && sourceEvent.ResourceIds.Count > 1)
                {
                    this.listResources.SelectedIndex = 1;
                }

            }
        }

        protected override void LoadResources()
        {
            base.LoadResources();

            if (this.schedulerData == null)
            {
                return;
            }

            ISchedulerStorage<IResource> resourceStorage = this.schedulerData.GetResourceStorage();
            this.listResources.Items.Clear();

            if (this.schedulerData.GroupType == GroupType.None)
            {
                RadListDataItem item = new RadListDataItem("None");
                item.Value = -1;
                this.listResources.Items.Add(item);
            }

            foreach (IResource resource in resourceStorage)
            {
                if (resource.Visible)
                {
                    RadListDataItem item = new RadListDataItem(resource.Name);
                    item.Value = resource.Id;
                    this.listResources.Items.Add(item);
                }
            }

            if (this.listResources.Items.Count > 0)
            {
                this.listResources.SelectedIndex = 0;
            }
        }

        protected override void ApplySettingsToEvent(IEvent targetEvent)
        {
            base.ApplySettingsToEvent(targetEvent);

            if (this.listResources.SelectedIndex >= 0)
            {
                RadListDataItem selectedItem = listResources.Items[this.listResources.SelectedIndex];

                if (selectedItem.Text.Equals("None"))
                {
                    if (targetEvent.ResourceIds.Count > 0)
                    {
                        targetEvent.ResourceIds.Clear();
                    }
                }
                else
                {
                    targetEvent.ResourceIds.Clear();

                    foreach (RadListDataItem item in listResources.SelectedItems)
                    {
                        EventId resourceId = item.Value as EventId;
                        if (!targetEvent.ResourceIds.Contains(resourceId))
                        {
                            targetEvent.ResourceIds.Add(resourceId);
                        }
                    }
                }
            }
        }

        private void InitializeComponent()
        {
            this.listResources = new Telerik.WinControls.UI.RadListControl();
            ((System.ComponentModel.ISupportInitialize)(this.labelSubject)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelLocation)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelBackground)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.txtSubject)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.txtLocation)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.cmbBackground)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelStartTime)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelEndTime)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.dateStart)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.timeStart)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.dateEnd)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.timeEnd)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.chkAllDay)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelResource)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.cmbResource)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelStatus)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.cmbShowTimeAs)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.textBoxDescription)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonOK)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonCancel)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonDelete)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonRecurrence)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.errorProvider)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.radSeparator1)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.radSeparator2)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.radSeparator3)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this.listResources)).BeginInit();
            ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
            this.SuspendLayout();
            // 
            // cmbBackground
            // 
            // 
            // 
            // 
            this.cmbBackground.RootElement.AutoSizeMode = Telerik.WinControls.RadAutoSizeMode.WrapAroundChildren;
            this.cmbBackground.Size = new System.Drawing.Size(180, 20);
            // 
            // dateStart
            // 
            this.dateStart.Value = new System.DateTime(2011, 5, 17, 11, 22, 29, 614);
            // 
            // timeStart
            // 
            this.timeStart.Value = new System.DateTime(2011, 5, 17, 11, 22, 29, 614);
            // 
            // dateEnd
            // 
            this.dateEnd.Value = new System.DateTime(2011, 5, 17, 11, 22, 29, 614);
            // 
            // timeEnd
            // 
            this.timeEnd.Value = new System.DateTime(2011, 5, 17, 11, 22, 29, 614);
            // 
            // chkAllDay
            // 
            // 
            // 
            // 
            this.chkAllDay.RootElement.ForeColor = System.Drawing.Color.FromArgb(((int)(((byte)(5)))), ((int)(((byte)(5)))), ((int)(((byte)(5)))));
            // 
            // cmbResource
            // 
            this.cmbResource.Location = new System.Drawing.Point(345, 169);
            this.cmbResource.Size = new System.Drawing.Size(180, 20);
            this.cmbResource.Visible = false;
            // 
            // cmbShowTimeAs
            // 
            // 
            // 
            // 
            this.cmbShowTimeAs.RootElement.AutoSizeMode = Telerik.WinControls.RadAutoSizeMode.WrapAroundChildren;
            this.cmbShowTimeAs.Size = new System.Drawing.Size(178, 20);
            // 
            // textBoxDescription
            // 
            this.textBoxDescription.Location = new System.Drawing.Point(7, 322);
            // 
            // 
            // 
            this.textBoxDescription.RootElement.StretchVertically = true;
            this.textBoxDescription.Size = new System.Drawing.Size(519, 116);
            // 
            // buttonOK
            // 
            this.buttonOK.Location = new System.Drawing.Point(6, 444);
            // 
            // buttonCancel
            // 
            this.buttonCancel.Location = new System.Drawing.Point(87, 444);
            // 
            // buttonDelete
            // 
            this.buttonDelete.Location = new System.Drawing.Point(168, 444);
            // 
            // buttonRecurrence
            // 
            this.buttonRecurrence.Location = new System.Drawing.Point(249, 444);
            // 
            // radSeparator3
            // 
            this.radSeparator3.Location = new System.Drawing.Point(6, 306);
            // 
            // listResources
            // 
            this.listResources.CaseSensitiveSort = true;
            this.listResources.ItemHeight = 18;
            this.listResources.Location = new System.Drawing.Point(79, 169);
            this.listResources.Name = "listResources";
            this.listResources.SelectionMode = System.Windows.Forms.SelectionMode.MultiSimple;
            this.listResources.Size = new System.Drawing.Size(180, 131);
            this.listResources.TabIndex = 25;
            this.listResources.Text = "radListControl1";
            // 
            // MultipleResourcesDialog
            // 
            this.ClientSize = new System.Drawing.Size(537, 472);
            this.Controls.Add(this.listResources);
            this.Name = "MultipleResourcesDialog";
            // 
            // 
            // 
            this.RootElement.ApplyShapeToControl = true;
            this.RootElement.MinSize = new System.Drawing.Size(539, 365);
            this.Text = "Edit Appointment";
            this.Controls.SetChildIndex(this.listResources, 0);
            this.Controls.SetChildIndex(this.radSeparator3, 0);
            this.Controls.SetChildIndex(this.cmbResource, 0);
            this.Controls.SetChildIndex(this.labelSubject, 0);
            this.Controls.SetChildIndex(this.labelLocation, 0);
            this.Controls.SetChildIndex(this.labelBackground, 0);
            this.Controls.SetChildIndex(this.txtSubject, 0);
            this.Controls.SetChildIndex(this.txtLocation, 0);
            this.Controls.SetChildIndex(this.cmbBackground, 0);
            this.Controls.SetChildIndex(this.labelStartTime, 0);
            this.Controls.SetChildIndex(this.labelEndTime, 0);
            this.Controls.SetChildIndex(this.dateStart, 0);
            this.Controls.SetChildIndex(this.timeStart, 0);
            this.Controls.SetChildIndex(this.dateEnd, 0);
            this.Controls.SetChildIndex(this.timeEnd, 0);
            this.Controls.SetChildIndex(this.chkAllDay, 0);
            this.Controls.SetChildIndex(this.labelResource, 0);
            this.Controls.SetChildIndex(this.labelStatus, 0);
            this.Controls.SetChildIndex(this.cmbShowTimeAs, 0);
            this.Controls.SetChildIndex(this.textBoxDescription, 0);
            this.Controls.SetChildIndex(this.buttonOK, 0);
            this.Controls.SetChildIndex(this.buttonCancel, 0);
            this.Controls.SetChildIndex(this.buttonDelete, 0);
            this.Controls.SetChildIndex(this.buttonRecurrence, 0);
            this.Controls.SetChildIndex(this.radSeparator1, 0);
            this.Controls.SetChildIndex(this.radSeparator2, 0);
            ((System.ComponentModel.ISupportInitialize)(this.labelSubject)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelLocation)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelBackground)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.txtSubject)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.txtLocation)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.cmbBackground)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelStartTime)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelEndTime)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.dateStart)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.timeStart)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.dateEnd)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.timeEnd)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.chkAllDay)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelResource)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.cmbResource)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.labelStatus)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.cmbShowTimeAs)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.textBoxDescription)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonOK)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonCancel)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonDelete)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.buttonRecurrence)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.errorProvider)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.radSeparator1)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.radSeparator2)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.radSeparator3)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this.listResources)).EndInit();
            ((System.ComponentModel.ISupportInitialize)(this)).EndInit();
            this.ResumeLayout(false);
            this.PerformLayout();

        }
    }
}
